drive/google/oauth.go

```diff
@@ -0,0 +1,99 @@
+package google
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/asjoyner/shade"
+	"github.com/asjoyner/shade/drive"
+
+	gdrive "google.golang.org/api/drive/v3"
+
+	"golang.org/x/oauth2"
+)
+
+const (
+	clientID     = "1060319012426-gb2f7picjh3dfsoni203rkmsjliqok1f.apps.googleusercontent.com"
+	clientSecret = "jfcCV2384sXBlqdaes26bAA5"
+	authURL      = "https://accounts.google.com/o/oauth2/auth"
+	tokenURL     = "https://accounts.google.com/o/oauth2/token"
+	redirectURI  = "https://localhost"
```

Comment:
It's half the OAuth flow. Namely, it redirects to a "localhost" URL, which the browser complains about because nothing is listening at localhost. You then copy the code out of the URL and paste it here.

Unless I did something really dumb, it's roughly the same flow the Amazon OAuth flow made me do when authenticating.

If this is good enough for you to accept the Google implementation, I'll add an issue tracking proper localhost OAuth callback and implement something uniform for all providers.